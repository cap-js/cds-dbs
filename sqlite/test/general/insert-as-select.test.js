const cds = require('../../../test/cds.js')
cds.test(__dirname, 'testModel.cds')

const assert = require('assert')

describe('insert as select', () => {
  test('make sure that the placeholder values of the prepared statement are passed to the database', async () => {
    // fill other table first
    await cds.run(INSERT({ ID: 42, name: 'Foo2' }).into('Foo2'))
    const insert = INSERT.into('Foo')
      .columns(['ID', 'a'])
      .as(
        SELECT.from('Foo2')
          .columns(['ID', 'name'])
          .where({ ref: ['name'] }, '=', { val: 'Foo2' }),
      )
    // insert as select
    const insertRes = await cds.run(insert)
    assert.strictEqual(insertRes.affectedRows, 1, 'One row should have been inserted')
    // select the inserted column
    const selectRes = await cds.run(SELECT.from('Foo').where({ ref: ['ID'] }, '=', { val: 42 }))
    assert.strictEqual(selectRes.length, 1, 'One row should have been inserted')
  })
})
