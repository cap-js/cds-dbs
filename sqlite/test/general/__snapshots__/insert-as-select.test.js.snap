// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"exec - swFX3mSaTpBcIO4VaMo4btiduC8="`;

exports[` 2`] = `"{"name":":memory:","open":true,"inTransaction":true,"readonly":false,"memory":true}"`;

exports[` 3`] = `"prepare - lS5P8r66aSBxOeSioFcRW/CO//g="`;

exports[` 4`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 5`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 6`] = `"prepare - Ui1MOXnTC6zIr4GLcZwkOSwnM10="`;

exports[` 7`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 8`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 9`] = `"prepare - OsymdlgtyQB+Fg/F+pOObPFNqlM="`;

exports[` 10`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 11`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 12`] = `"prepare - 5kUYYfbXDY3POU3hSUHzqEJ3W6w="`;

exports[` 13`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 14`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 15`] = `"prepare - ZShdnN+t61YSy57mTrvXwkzCYAU="`;

exports[` 16`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 17`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 18`] = `"prepare - xl/W7pjIQ7iLS6PT9Gu9KtZPbT8="`;

exports[` 19`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 20`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 21`] = `"prepare - P1SiILO9fbcNOYBDQtvEJ1wtNUg="`;

exports[` 22`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 23`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 24`] = `"prepare - uND1ro1gJHs2YgnVqj0OuoZdb34="`;

exports[` 25`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 26`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 27`] = `"prepare - y0ZUXMKPZ4eUaunTFebCQtDA5Sc="`;

exports[` 28`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 29`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 30`] = `"prepare - o90RA7P1/wpKwl20WJp+n0ScoN8="`;

exports[` 31`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 32`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 33`] = `"prepare - j6BQlowBPGDzG4y8iW9108/ebJ4="`;

exports[` 34`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 35`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 36`] = `"prepare - +NsW9LSCFBM0up5qNORr6qVZATo="`;

exports[` 37`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 38`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 39`] = `"prepare - uiziVov1Xk4gW9LhOhzmlSKem+0="`;

exports[` 40`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 41`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 42`] = `"prepare - lRoxrLxTxTAC40iYX2rJ5ffX6t0="`;

exports[` 43`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 44`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 45`] = `"prepare - 4e0+fytAbfIIAmIEKRNEk6cWeRo="`;

exports[` 46`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 47`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 48`] = `"prepare - Ce5YJylmKakT6ewHJp2Yfvk3smY="`;

exports[` 49`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 50`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 51`] = `"prepare - GYymgaKUDr3Jwsl2GBuuohdrLZM="`;

exports[` 52`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 53`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 54`] = `"prepare - FodT+w5KnN6oqW916eRVAMf/giI="`;

exports[` 55`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 56`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 57`] = `"prepare - AW68aGeaXBrgvXpf1/Tgkvzlzi0="`;

exports[` 58`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 59`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 60`] = `"prepare - Tn1SyuhLyKiqEbPnAP29QuKxyL4="`;

exports[` 61`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 62`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 63`] = `"prepare - qsKcVDio5ZlRlsptZdNqDvlXUwY="`;

exports[` 64`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 65`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 66`] = `"prepare - +G3rvtLryr5ODoTCgzqfOv7IivE="`;

exports[` 67`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 68`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 69`] = `"prepare - xVc954UxsWKLKbMu5JlwT1+2ZmE="`;

exports[` 70`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 71`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 72`] = `"prepare - 0pGthlgDBaa4Nf6MPnue1QT7wvg="`;

exports[` 73`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 74`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 75`] = `"prepare - Jdd2Tkk9txk3+2Aqx9IydDvEipE="`;

exports[` 76`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 77`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 78`] = `"prepare - JzekHJZKCH+d4VHWcSUdz7JEsRo="`;

exports[` 79`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 80`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 81`] = `"prepare - hJE/Sbm+sHGat5W+HG/GeiNIe2E="`;

exports[` 82`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 83`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 84`] = `"exec - mvfei0i6/P7jaJW/2LTxQWnJXeI="`;

exports[` 85`] = `"{"name":":memory:","open":true,"inTransaction":false,"readonly":false,"memory":true}"`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 1`] = `"exec - swFX3mSaTpBcIO4VaMo4btiduC8="`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 2`] = `"{"name":":memory:","open":true,"inTransaction":true,"readonly":false,"memory":true}"`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 3`] = `"prepare - z1MV085VALJug3bgv14aePEj/xg="`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 4`] = `"run - AdrrSCc/saDItpB/p/7vkX1oJGc="`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 5`] = `"{"changes":1,"lastInsertRowid":42}"`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 6`] = `"exec - mvfei0i6/P7jaJW/2LTxQWnJXeI="`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 7`] = `"{"name":":memory:","open":true,"inTransaction":false,"readonly":false,"memory":true}"`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 8`] = `"exec - swFX3mSaTpBcIO4VaMo4btiduC8="`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 9`] = `"{"name":":memory:","open":true,"inTransaction":true,"readonly":false,"memory":true}"`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 10`] = `"prepare - X8EgTqiD7IloqgOTRT6GNc6N4RM="`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 11`] = `"run - JE2c6+sOU4xTtxNKvI2ylxw7hIw="`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 12`] = `"{"changes":1,"lastInsertRowid":42}"`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 13`] = `"exec - mvfei0i6/P7jaJW/2LTxQWnJXeI="`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 14`] = `"{"name":":memory:","open":true,"inTransaction":false,"readonly":false,"memory":true}"`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 15`] = `"exec - swFX3mSaTpBcIO4VaMo4btiduC8="`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 16`] = `"{"name":":memory:","open":true,"inTransaction":true,"readonly":false,"memory":true}"`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 17`] = `"prepare - dQ0nANJzTd3TR4KOSVi7ocdxuXk="`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 18`] = `"all - kTjdZfiiPc+sbBOps1XCuu0aYrM="`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 19`] = `"[{"_json_":"{\\"ID\\":42,\\"a\\":\\"Foo2\\",\\"b\\":null,\\"c\\":null,\\"x\\":null}"}]"`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 20`] = `"exec - mvfei0i6/P7jaJW/2LTxQWnJXeI="`;

exports[`insert as select make sure that the placeholder values of the prepared statement are passed to the database 21`] = `"{"name":":memory:","open":true,"inTransaction":false,"readonly":false,"memory":true}"`;
